src/osd/ReplicatedBackend.cc

```diff
@@ -1259,8 +1265,13 @@ void ReplicatedBackend::sub_op_modify_applied(RepModifyRef rm)
   // send ack to acker only if we haven't sent a commit already
   if (ack) {
     ack->set_priority(CEPH_MSG_PRIO_HIGH); // this better match commit priority!
-    get_parent()->send_message_osd_cluster(
-      rm->ackerosd, ack, get_osdmap()->get_epoch());
+    if (!isBlueStore) {
+      get_parent()->send_message_osd_cluster(
+        rm->ackerosd, ack, get_osdmap()->get_epoch());
+    }
+    else {
+      ack->put();
+    }
```

Comment:
We shouldn't need a conditional on the lower-level backend here.  Why is bluestore different than filestore?  Is it because the bluestore ack callback happens before the commit callback?  In that case, the client should not request an ack at all (which we just changed with librados and librbd, btw).  Can you share a bit more about the motivation?